frontend/src/lib/components/ModuleStep.svelte

```diff
@@ -1,173 +1,89 @@
 <script lang="ts">
-	import type { Schema } from '$lib/common'
-	import { FlowModuleValue, type Flow, type FlowModule } from '$lib/gen'
-	import { inferArgs } from '$lib/infer'
-	import { loadSchema as UloadSchema } from '$lib/scripts'
-	import { DENO_INIT_CODE, PYTHON_INIT_CODE } from '$lib/script_helpers'
-	import { addPreviewResult, previewResults, workspaceStore } from '$lib/stores'
-	import {
-		buildExtraLib,
-		emptySchema,
-		objectToTsType,
-		schemaToObject,
-		schemaToTsType
-	} from '$lib/utils'
-	import { language, mehO } from 'svelte-awesome/icons'
+	import { FlowModuleValue, type FlowModule } from '$lib/gen'
+	import { addPreviewResult, previewResults } from '$lib/stores'
+	import { buildExtraLib, objectToTsType, schemaToObject, schemaToTsType } from '$lib/utils'
+	import { get } from 'svelte/store'
 	import Editor from './Editor.svelte'
 	import FlowPreview from './FlowPreview.svelte'
-	import RadioButton from './RadioButton.svelte'
+	import FlowInputs from './flows/FlowInputs.svelte'
+	import FlowModuleHeader from './flows/FlowModuleHeader.svelte'
+	import {
+		createInlineScriptModule,
+		flowStore,
+		loadSchema,
+		pickScript,
+		schemasStore
+	} from './flows/flowStore'
 	import SchemaForm from './SchemaForm.svelte'
-	import ScriptPicker from './ScriptPicker.svelte'
 
-	export let flow: Flow
 	export let i: number
 	export let mod: FlowModule
 	export let args: Record<string, any> = {}
 
-	export let schemas: Schema[] = []
-	export let schemaForms: (SchemaForm | undefined)[] = []
-
-	let editor: Editor
-	let flowPreview: FlowPreview
-
-	$: previousSchema = i === 0 ? schemaToObject(flow.schema) : $previewResults[i]
+	const flow = get(flowStore)
 
+	$: schema = $schemasStore[i]
```

Comment:
There might be some cases where you remove the step before this component is removed and so you want to check that schemaStore length is big enough for index i, I think.

Reply:
@fatonramadani still need fix i think